### PR TITLE
Rybrande/return cookie token

### PR DESCRIPTION
**Pull request by ryanbrandenburg**
Always return the cookie token, weather it's old or new.

Needed for my current MusicStore PR.

### CODE REVIEW

File: src/Microsoft.AspNet.Antiforgery/DefaultAntiforgery.cs [hunk 1]
```diff
@@ -243,6 +247,8 @@ private class AntiforgeryTokenSetInternal
             public AntiforgeryToken FormToken { get; set; }
 
             public AntiforgeryToken CookieToken { get; set; }
+
+            public bool CookieTokenIsNew { get; set; }
```
Comment (rynowak) [inline src/Microsoft.AspNet.Antiforgery/DefaultAntiforgery.cs]: Nit: `IsNewCookieToken`

**Comment by rynowak** [comment]
Can I get a summary of the problem solved by this? The code changes are simple and seem safe. What's the overall goal?

File: test/Microsoft.AspNet.Antiforgery.Test/DefaultAntiforgeryTest.cs [hunk 2]
```diff
@@ -313,11 +313,30 @@ public void GetTokens_ExistingValidCookieToken_GeneratesANewFormToken()
...
+        {
+            var context = CreateMockContext(
+                new AntiforgeryOptions(),
+                useOldCookie: true,
+                isOldCookieValid: true);
+            var antiforgery = GetAntiforgery(context);
+
+            var tokenSet = antiforgery.GetAndStoreTokens(context.HttpContext);
+            
+            //Strict mock would have complained if SaveCookieToken was called.
```
Comment (rynowak) [inline test/Microsoft.AspNet.Antiforgery.Test/DefaultAntiforgeryTest.cs]: CS01092: Comment not useful at DefaultAntiforgeryTest.cs (332:12)

File: test/Microsoft.AspNet.Antiforgery.Test/DefaultAntiforgeryTest.cs [hunk 3]
```diff
@@ -313,11 +313,30 @@ public void GetTokens_ExistingValidCookieToken_GeneratesANewFormToken()
             var tokenset = antiforgery.GetTokens(context.HttpContext);
 
             // Assert
-            Assert.Null(tokenset.CookieToken);
+            Assert.NotNull(tokenset.CookieToken);
```
Comment (rynowak) [inline test/Microsoft.AspNet.Antiforgery.Test/DefaultAntiforgeryTest.cs]: These `Assert.NotNulls` really aren't needed.

File: test/Microsoft.AspNet.Antiforgery.Test/DefaultAntiforgeryTest.cs [hunk 4]
```diff
@@ -313,11 +313,30 @@ public void GetTokens_ExistingValidCookieToken_GeneratesANewFormToken()
...
+            var antiforgery = GetAntiforgery(context);
+
+            var tokenSet = antiforgery.GetAndStoreTokens(context.HttpContext);
+            
+            //Strict mock would have complained if SaveCookieToken was called.
+
+            Assert.NotNull(tokenSet.CookieToken);
+            Assert.Equal("serialized-old-cookie-token", tokenSet.CookieToken);
+            Assert.Equal("serialized-form-token", tokenSet.FormToken);
+        }
```
Comment (rynowak) [inline test/Microsoft.AspNet.Antiforgery.Test/DefaultAntiforgeryTest.cs]: Where are the unit tests verifying that fancy new property?

**Comment by rynowak** [comment]
:watch:

**Comment by Eilon** [comment]
@blowdart because @blowdart and because anti-forgery. Just making sure you're cool with this.

**Comment by ryanbrandenburg** [comment]
@rynowak the problem being solved here is that if there was already a cookie token existing when you call GetTokens it doesn't give you a cookie token, only a new form token. Since calling GetTokens is apparently how we want people to do Anti-forgery for AJAX calls it is important that it give you both of these.

File: test/Microsoft.AspNet.Antiforgery.Test/DefaultAntiforgeryTest.cs (hunk 4 above)
Comment (ryanbrandenburg) [inline test/Microsoft.AspNet.Antiforgery.Test/DefaultAntiforgeryTest.cs]: The above test was intended to be the test of the property. If the CookieToken is not new, then we shouldn't be trying to save it again. I've added a test to ensure we save it when we actually do create a new token as well, will put that up with the update.

**Comment by ryanbrandenburg** [comment]
:up::date:

**Comment by rynowak** [comment]
Groovy :shipit:

**Comment by blowdart** [comment]
Damnit, I can't find anything to pick on. This sets a worrying precedent for new Ryan. :shipit:

**Comment by Eilon** [comment]
@ryanbrandenburg FYI that was a huge compliment from @blowdart :smile:

**Pull request merged**